arches_querysets/rest_framework/serializers.py

```diff
@@ -488,10 +492,33 @@ def create(self, validated_data):
             validated_data["data"] = {}
         validated_data["__request"] = self.context["request"]
         validated_data["__as_representation"] = True
-        with transaction.atomic():
+        resource, resource_created = self.create_resource_if_missing(
+            validated_data, entry_node=qs._entry_node
+        )
+        try:
             created = super().create(validated_data)
+        except:
+            if resource_created:
+                # Manually manage failures instead of using transaction.atomic(), see:
+                # https://github.com/archesproject/arches/issues/12318
+                # Don't want to run model delete() which *creates* edit log entries.
+                EditLog.objects.filter(resourceinstanceid=resource.pk).delete()
+                Resource.objects.filter(pk=resource.pk).delete()
+            raise
         return created
 
+    def create_resource_if_missing(self, validated_data, entry_node):
+        if instance := validated_data.get("resourceinstance"):
```

Comment:
I'm not sure how validated_data["resourceinstance"] is populated but any "truthy" value here will work.  Are we confident that "instance" will always be a "Resource" instance object?

Reply:
Yeah, that's one of the biggest usability wins provided by DRF -- you can't access the variable `validated_data` until the validation has run, and once it has, you can depend on it being the type defined by your serializer field, which here is `ResourceInstance | None`.

Reply:
You can think of the serializer as a form. The form is in charge of coercing the data to the expected type.

Reply:
> Yeah, that's one of the biggest usability wins provided by DRF -- you can't access the variable `validated_data` until the validation has run, and once it has, you can depend on it being the type defined by your serializer field, which here is `ResourceTileTree | None`.

Thanks for the explanation @jacobtylerwalls !  That's helpful to know